### PR TITLE
Add PF2e typed modifier stacking

### LINKED ISSUE #81: [Rules] Add typed PF2e modifier stacking
## Context
Discovered during PR #79 review for issue #76. Ranged Strike support currently applies MAP, range penalties, and cover directly in `Strike.DamageEvaluate`, while imported NPC weapon action values are carried as a total attack modifier.

PF2e modifiers need type-aware stacking: item, status, and circumstance modifiers do not stack with other modifiers of the same type, while untyped/flat modifiers generally do. Cover is a circumstance AC bonus, and conditions such as off-guard/flat-footed are circumstance AC penalties.

## Current Behavior
Combat math currently stores broad totals such as `CreatureComponent.attackBonus`, `Strike.AttackModifierOverride`, and `StrikeTargetResult.CoverAcBonus`. The calculation does not preserve modifier type/source, so future effects could accidentally stack incorrectly.

## Expected Behavior
Attack rolls, AC, saves, DCs, and other PF2e checks should resolve through a shared typed modifier model that applies PF2e stacking rules consistently.

## Scope
- Add a reusable modifier model with value, type, source, and target statistic.
- Apply highest same-type item/status/circumstance bonus or penalty as appropriate.
- Preserve support for untyped/flat modifiers that stack.
- Route Strike MAP, range penalty, cover, and imported NPC weapon action modifiers through the model.
- Keep deterministic tests around edge cases such as cover plus other circumstance AC modifiers.

## Acceptance Criteria
- [ ] Unit tests cover same-type bonus and penalty non-stacking.
- [ ] Unit tests cover different-type modifiers stacking.
- [ ] Strike attack rolls use the shared modifier resolver for MAP, range, and cover.
- [ ] Existing EditMode and PlayMode combat tests still pass.

## Notes
Related PR: #79
Related files: `Assets/Scripts/Combat/Actions/Attacks/Strike.cs`, `Assets/Scripts/Creature/CreatureComponent.cs`, `Assets/Scripts/Grid/StrikeTargeting.cs`.

### PR DESCRIPTION
**Pull request by clausman**
Closes #81

## Summary
- Added a shared PF2e typed modifier model/resolver for untyped, circumstance, item, and status modifiers.
- Routed Strike attack and AC math through the resolver for base attack totals, MAP, range penalties, cover, and current creature modifiers.
- Added CreatureComponent resolver helpers for attack rolls, AC, saves, skills, initiative, and DCs while preserving existing serialized stat fields and imported JSON totals.
- Documented the design choices and linked AoN rules references in `docs/pf2e-modifier-stacking.md`.

## Testing
- EditMode: 26 passed, 0 failed (`TestResults/EditModeResults.xml`)
- PlayMode: 29 passed, 0 failed (`TestResults/PlayModeResults.xml`)

## Visual Evidence
- Not applicable: rules/math and documentation change only.

### CODE REVIEW

File: Assets/Scripts/Creature/CreatureComponent.cs [hunk 1]
```diff
@@ -151,6 +155,176 @@ public int GetAttackBonusForWeapon(EquipmentWeapon weapon)
...
+        {
+            List<Pf2eModifier> modifiers = new()
+            {
+                new Pf2eModifier(baseDc, Pf2eModifierType.Untyped, "Base DC", Pf2eStatistic.DifficultyClass)
+            };
+            AddRuntimeModifiers(modifiers, additionalModifiers);
+            return Pf2eModifierResolver.Resolve(modifiers, Pf2eStatistic.DifficultyClass);
+        }
+
+        private Pf2eModifierResolution ResolveSave(int baseSave, string source, Pf2eStatistic statistic, IEnumerable<Pf2eModifier> additionalModifiers)
```
Comment (clausman) [inline Assets/Scripts/Creature/CreatureComponent.cs:280]: What is the purpose of `ResolveSave` when there are separate will, reflex, and fortitude saves above?

File: Assets/Scripts/Creature/CreatureComponent.cs [hunk 2]
```diff
@@ -151,6 +155,176 @@ public int GetAttackBonusForWeapon(EquipmentWeapon weapon)
...
+            {
+                if (string.Equals(armorBonus.category, category, StringComparison.OrdinalIgnoreCase))
+                    return armorBonus.bonus;
+            }
+            return 0;
+        }
+
+        private void AddConditionModifiers(List<Pf2eModifier> modifiers, Pf2eStatistic statistic)
+        {
+            if (statistic == Pf2eStatistic.ArmorClass && HasActiveCondition("off-guard", "flat-footed"))
```
Comment (clausman) [inline Assets/Scripts/Creature/CreatureComponent.cs]: I'm worried about this becoming a monolithic if block in the future if it needs to cover all conditions for all statistics

File: Assets/Scripts/Rules/Pf2eModifiers.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        ArmorClass,
+        FortitudeSave,
+        ReflexSave,
+        WillSave,
+        SkillCheck,
+        Initiative,
+        DifficultyClass
+    }
+
+    public static class Pf2eRuleReferences
```
Comment (clausman) [inline Assets/Scripts/Rules/Pf2eModifiers.cs]: It seems like these `Pf2eRuleReferences` should be in a code comment rather than strings here unless you think these are valuable to surface programatically for something

File: docs/pf2e-modifier-stacking.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+The resolver filters by target statistic. Untyped modifiers stack. For each typed category, the highest bonus applies and lower bonuses of the same type are suppressed; the worst penalty applies and milder penalties of the same type are suppressed. A same-type bonus and same-type penalty can both apply.
+
+## Current Integration
+
+Current serialized creature fields remain compatibility inputs:
+
+- Imported NPC attack totals and weapon action bonuses are treated as untyped base attack modifiers because the current JSON does not decompose them by PF2e modifier type.
+- Imported or manually assigned AC is treated as an untyped base AC unless the creature has equipped armor data available.
+- Equipped armor contributes its armor AC as an item modifier, while Dex and proficiency contributions remain untyped base components.
```
Comment (clausman) [inline docs/pf2e-modifier-stacking.md]: I don't think equipped armor is an item modifier, I think only armor runes you get at higher levels are item modifiers. Double check the rules on this one.

File: Assets/Scripts/Creature/CreatureComponent.cs (hunk 2 above)
Comment (clausman) [reply Assets/Scripts/Creature/CreatureComponent.cs]: Agreed. I refactored this so `CreatureComponent` no longer owns a runtime modifier list or condition modifier mapping. It now gathers `IPf2eModifierProvider` components from the GameObject. `Conditions` provides condition modifiers through `ConditionModifierRules`, and `Pf2eModifierCollection` is only a generic simple-provider escape hatch for temporary sources without dedicated components yet.

File: Assets/Scripts/Rules/Pf2eModifiers.cs (hunk 3 above)
Comment (clausman) [reply Assets/Scripts/Rules/Pf2eModifiers.cs]: Agreed. I removed `Pf2eRuleReferences` and the `RulesReference` field from `Pf2eModifier`. The AoN links now live in code comments, test comments, and the design doc unless we later have a real runtime need to surface provenance.

File: Assets/Scripts/Creature/CreatureComponent.cs (hunk 1 above)
Comment (clausman) [reply Assets/Scripts/Creature/CreatureComponent.cs:280]: Kept the public Fortitude/Reflex/Will methods, but added a short comment on the private helper. It is only there to share the identical save-resolution path while preserving separate named entrypoints for callers.

File: docs/pf2e-modifier-stacking.md (hunk 4 above)
Comment (clausman) [reply docs/pf2e-modifier-stacking.md]: I double-checked this against AoN. The armor rules say the AC Bonus value is the armor item bonus to AC: https://2e.aonprd.com/Rules.aspx?ID=2166. I kept the implementation as an item modifier and updated the doc/code comment to cite that rule directly.
